angel_system/ptg_eval/common/discretize_data.py

```diff
@@ -0,0 +1,132 @@
+import numpy as np
+import logging
+
+log = logging.getLogger("ptg_eval_common")
+
+def discretize_data_to_windows(labels, gt, detections, time_window, uncertainty_pad):
+    """
+    :param labels: Array of class labels (str)
+    :param gt: Pandas dataframe of the ground truth
+    :param detections: Pandas dataframe of all detections per class
+    :param time_window: Time in seconds to discretize the whole time span by
+    :param uncertainty_pad: Time in seconds to pad ground truth detections by. Detections 
+        within the windows ground truth + or - uncertainty pad will not be scored
+
+    :return: Tuple(
+        ground truth mask - Matrix of size (number of valid time windows x number classes) 
+            where True indicates a true class example, False inidcates a false class example. 
+            There should only be one True value per row, 
+        window class scores - Matrix of size (number of valid time windows x number classes)
+            filled with the max confidence score per class for any detections in the time window
+        )
+    """
+    # ============================
+    # Split by time window
+    # ============================
+    # Get time ranges
+    assert time_window > uncertainty_pad, (
+        "Time window must be longer than the uncertainty pad"
+    )
+    min_start_time = min(gt['start'].min(), detections['start'].min())
+    max_end_time = max(gt['end'].max(), detections['end'].max())
+    dt = time_window
+    time_windows = np.arange(min_start_time, max_end_time, time_window)
+
+    if time_windows[-1] < max_end_time:
+        time_windows = np.append(time_windows, time_windows[-1] + time_window)
+    time_windows = list(zip(time_windows[:-1], time_windows[1:]))
+    time_windows = np.array(time_windows)
+    window_class_scores = np.zeros((len(time_windows), len(labels)),
+                                     dtype=float)
+
+
+    def get_time_wind_range(start, end):
```

Comment:
probably worth pulling this outside of this function.

Reply:
actually, maybe hold off on that since it uses other local variables defined inside this function.